scipy/stats/tests/test_distributions.py

```diff
@@ -371,6 +371,195 @@ def test_pdf_zero(self):
+            0.000941920705790324
+            ])
+
+        lmbda = 2
+        alpha = 2
+        beta = 1
+        delta = 1.5
+        mu = 0.5
+
+        alpha_hat = alpha * delta
+        beta_hat = beta * delta
+
+        x = np.linspace(-10, 10, 10)
+        gh = stats.genhyperbolic(
+            p=lmbda,
+            a=alpha_hat,
+            b=beta_hat,
+            loc=mu,
+            scale=delta
+            )
+
+        assert_allclose(gh.pdf(x), vals_R, atol=1e-9)
+
+    def test_cdf_r(self):
+        # test against R package GeneralizedHyperbolic
+        # q <- seq(-10, 10, length.out = 10)
+        # GeneralizedHyperbolic::pghyp(
+        #   q = q, lambda = 2, alpha = 2, beta = 1, delta = 1.5, mu = 0.5
+        # )
+        vals_R = np.array([
+            1.01881590921421e-13, 6.13697274983578e-11, 3.37504977637992e-08,
+            1.55258698166181e-05, 0.00447005453832497, 0.228935323956347,
+            0.755759458895243, 0.953061062884484, 0.992598013917513,
+            0.998942646586662
+            ])
+
+        lmbda = 2
+        alpha = 2
+        beta = 1
+        delta = 1.5
+        mu = 0.5
+
+        alpha_hat = alpha * delta
+        beta_hat = beta * delta
+
+        x = np.linspace(-10, 10, 10)
+        gh = stats.genhyperbolic(
+            p=lmbda,
+            a=alpha_hat,
+            b=beta_hat,
+            loc=mu,
+            scale=delta
+            )
+
+        assert_allclose(gh.cdf(x), vals_R, atol=1e-9, rtol=1e-5)
+
+    def test_moments_r(self):
+        # test against R package GeneralizedHyperbolic
+        # sapply(1:4,
+        #    function(x) GeneralizedHyperbolic::ghypMom(
+        #        order = x, lambda = 2, alpha = 2,
+        #        beta = 1, delta = 1.5, mu = 0.5,
+        #        momType = 'raw')
+        # )
+
+        vals_R = np.array([
+            2.36848366948115, 8.4739346779246,
+            37.8870502710066, 205.76608511485
+            ])
+
+        lmbda = 2
+        alpha = 2
+        beta = 1
+        delta = 1.5
+        mu = 0.5
+
+        alpha_hat = alpha * delta
+        beta_hat = beta * delta
+
+        vals_us = [
+            stats.genhyperbolic(p=lmbda, a=alpha_hat, b=beta_hat,
+                                loc=mu, scale=delta).moment(i)
+            for i in range(1, 5)
+            ]
+
+        assert_allclose(vals_us, vals_R, atol=1e-9, rtol=1e-8)
+
+    def test_rvs(self):
+        # KS test
+        # p = 1, alpha = 1, beta = 0
+        gh = stats.genhyperbolic(p=1, a=1, b=0)
+        _, p = stats.kstest(gh.rvs(size=1500, random_state=1234), gh.cdf)
+
+        assert_equal(p > 0.05, True)
+
+    def test_pdf_t(self):
+        # Test Against T-Student with 1 - 30 df
+        df = np.linspace(1, 30, 20)
+        # in principle alpha should be zero in practice for big lmbdas
+        # alpha cannot be too small else pdf does not integrate
+        alpha = np.float_power(df, 2)*np.finfo(np.float32).eps
```

Comment:
this might not be obvious for users who will probably try it with alpha=0

Reply:
I would say that if one wanted to fit a t-student, they would rather use stats.t instead of this. This is only to ensure the theory holds.

Reply:
yes, maybe one should even mention this in the docstring, e.g. 'For the distributions that are a special case such as Student's t, it is not recommended to rely on the implementation of genhyperbolic. To avoid potential numerical problems and for performance reasons, the methods of the specific distributions should be used. '

just a suggestion, maybe other reviewers have a different view.

Reply:
I think that's a good idea. Users are always reporting "bugs" like this so it would be nice if we had documentation to point to.

Reply:
It makes sense. Will amend the docstring.